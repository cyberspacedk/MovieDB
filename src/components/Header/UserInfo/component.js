import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'antd';
import styled from 'styled-components';

const UserStatus = styled.div`
  > span {
    color: #f300ff;
    margin-right: 10px;
    font-weight: 700;
  }
`;
const UserInfo = ({ username, userLogout }) => (
  <UserStatus>
    <span>{username}</span>
    <Button type="primary" onClick={userLogout}>
      Logout
    </Button>
  </UserStatus>
);

UserInfo.propTypes = {
  username: PropTypes.string.isRequired,
  userLogout: PropTypes.func.isRequired,
};

export default UserInfo;
