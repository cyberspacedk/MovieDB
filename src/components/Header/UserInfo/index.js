import ContainerUserInfo from './container';

export default ContainerUserInfo;
