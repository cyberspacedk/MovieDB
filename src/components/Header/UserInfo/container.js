import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import UserInfo from './component';
import { userLogout } from '../../../store/authentifiction/actions';
import {
  getUserLogin,
  isAuthentificated,
} from '../../../store/authentifiction/selectors';

// eslint-disable-next-line no-shadow
const ContainerUserInfo = ({ username, userLogout }) => {
  return <UserInfo userName={username} logout={userLogout} />;
};

const mstp = state => ({
  username: getUserLogin(state),
  sessionID: isAuthentificated(state),
});
const mdtp = {
  userLogout,
};

ContainerUserInfo.propTypes = {
  username: PropTypes.string.isRequired,
  userLogout: PropTypes.func.isRequired,
};

export default connect(
  mstp,
  mdtp,
)(ContainerUserInfo);
