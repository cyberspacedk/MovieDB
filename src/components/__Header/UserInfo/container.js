import { connect } from 'react-redux';
import UserInfo from './component';
import { userLogout } from '../../../store/authentifiction/actions';
import {
  getUserLogin,
  isAuthentificated,
} from '../../../store/authentifiction/selectors';

export const mstp = state => ({
  username: getUserLogin(state),
  sessionID: isAuthentificated(state),
});

const mdtp = {
  userLogout,
};

export default connect(
  mstp,
  mdtp,
)(UserInfo);
