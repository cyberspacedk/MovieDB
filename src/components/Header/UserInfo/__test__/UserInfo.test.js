import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import UserInfo from '../component';

describe('Component: UserInfo', () => {
  const props = {
    username: 'USERNAME',
    userLogout() {},
  };

  it('should match its snapshot. Display username', () => {
    const UserInfoComponent = shallow(<UserInfo {...props} />);
    expect(shallowToJson(UserInfoComponent)).toMatchSnapshot();
  });
});
