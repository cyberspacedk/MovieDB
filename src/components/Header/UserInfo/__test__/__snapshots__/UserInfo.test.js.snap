// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: UserInfo should match its snapshot. Display username 1`] = `
<styled.div>
  <span>
    USERNAME
  </span>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
    type="primary"
  >
    Logout
  </Button>
</styled.div>
`;
